### DISCUSSION TITLE
realtime push streaming memory leak

### DISCUSSION THREAD

**Discussion by zgr126** [aiortc/aiortc#704] [Q&A]
I tried to do a 1080p 60fps live video push stream at 6Mbps
```
options = {
        "video_size": "1920x1080",
        'framerate': '60',
        'rtbufsize': '160M',
        "profile": "baseline",
        'delay': '0',
        'zerolatency': '1',
        'preset': 'p1',
        'rc': 'vbr',
    }
```
But after pushing for a while (sometimes a few seconds), the network rate drops instantly to very low, the memory starts to grow and ffmpeg starts reporting errors
```
[dshow @ 0000020f91615180] real-time buffer [2K Camera] [video input] too full or near too full (75% of size: 
160000000 [rtbufsize parameter])! frame dropped!
[dshow @ 0000020f91615180] real-time buffer [2K Camera] [video input] too full or near too full (75% of size: 
160000000 [rtbufsize parameter])! frame dropped!
......
```
But when the bitrate is reduced to 2Mbps it basically doesn't happen again, so I think it's related to the network. I found line 343 in rtcrtpsender.py "await self.transport._send_rtp(packet_bytes)"
and detect the time, which is about
```
sendtime:0.03789806365966797
sendtime:0.004986286163330078
```

Is it possible that the time spent waiting to send in the for loop is taking up the encoding time? that it is not possible to continue encoding in 16ms and thus memory is overflowing?

**Comment by jlaine** [comment]
The errors you sent do not indicate a memory leak, they indicate you are not able to read from the camera fast enough. Most likely you are right, encoding is taking too much time.
